bundle/status/base_compliance_status_bundle.go

```diff
@@ -9,32 +9,32 @@ type PolicyGenericComplianceStatus struct {
 	UnknownComplianceClusters []string `json:"unknownComplianceClusters"`
 }
 
-// BaseClustersPerPolicyBundle is the base struct for clusters per policy bundle and contains the full state.
-type BaseClustersPerPolicyBundle struct {
-	Objects     []*PolicyGenericComplianceStatus `json:"objects"`
-	LeafHubName string                           `json:"leafHubName"`
-	Generation  uint64                           `json:"generation"`
-}
-
 // PolicyCompleteComplianceStatus struct holds information for (optimized) policy compliance status.
 type PolicyCompleteComplianceStatus struct {
 	PolicyID                  string   `json:"policyId"`
 	NonCompliantClusters      []string `json:"nonCompliantClusters"`
 	UnknownComplianceClusters []string `json:"unknownComplianceClusters"`
 }
 
+// BaseClustersPerPolicyBundle is the base struct for clusters per policy bundle and contains the full state.
+type BaseClustersPerPolicyBundle struct {
+	Objects     []*PolicyGenericComplianceStatus `json:"objects"`
+	LeafHubName string                           `json:"leafHubName"`
+	BundleVersion
```

Comment:
I'm sure it works, but asking out of curiosity (mainly for consistency around our code).
why don't you write this line as:
```
BundleVersion BundleVersion `json:bundleVersion`
```
will it work the same if you change this?
(same goes for all three structs)

Reply:
I thought it may mess up the marshaling/unmarshalling for senders that use this code and receivers that use the previous (with generation alone) so wanted to stay on the safe side and have the fields of BundleVersion exported on their own, but exporting the BundleVersion works too: https://play.golang.org/p/CrA7pKHzbgx

(done)

Reply:
@vMaroon your code sample actually proves that it's **NOT WORKING** when using the previous version.
if you noticed, when you print C, the int value is 0, not 42.

I've added another printing line to your sample, just to make sure both of us understand what's happening.
see it here: https://play.golang.org/p/bcpn8K71w9D
try to run it once as it and then another time only with `A` under struct `B`.

That being said, we do not have any requirement that we should support sender/receiver running incompatible versions or need to have backward compatability.
the work that we're doing is still WIP and not a product and it's a valid claim to say that both sides should use the same version of datatypes module - so I think your latest code should be left as is and merged.

waiting for your comment before merging in case I'm missing something.